### PR TITLE
[FIX] website_sale: avoid adding website on customer when not needed

### PR DESCRIPTION
**Pull request by rdeodoo**
Before this commit, when an address would be edited or created from checkout,
we would always add a website_id on it.
(values_postprocess -> _checkout_form_save)

That would be incorrect if the website has not 'specific user account' enabled.

It would lead to an error as the user would not be able to connect to another
website. Indeed, the billing address is stored on the partner directly. Since
the res.user website_id is a related to the partner one, the user would then be
restricted to that website.

Fixes https://github.com/odoo/odoo/issues/34308

### CODE REVIEW

**Comment by rdeodoo** [comment]
@JKE-be I removed from tools and adapted+moved it to website new file (the same that is added in higher version).
There is an occurence of `sale_order_id` in the MockRequest in `website`, do you want me to find a way to make it overridable so we override it in website_sale?
That's just for test but still, let me know what you decide.

File: addons/website_sale/tests/test_sale_process.py [hunk 1]
```diff
@@ -14,4 +18,32 @@ def test_02_admin_checkout(self):
...
+        })
+        self.env['sale.order.line'].create({
+            'product_id': product.id,
+            'name': 'Product A',
+            'order_id': so.id,
+        })
+
+        WebsiteSaleController = WebsiteSale()
+        with MockRequest(self.env, website=website, sale_order_id=so.id):
+            WebsiteSaleController.address(name='website_sale test user', email='email@email.email', street='ooo', city='ooo', country_id=18, submitted=1, partner_id=p.id)
```
Comment (JKE-be) [inline addons/website_sale/tests/test_sale_process.py]: search[], limit=1)

File: addons/website_sale/tests/test_sale_process.py [hunk 2]
```diff
@@ -14,4 +18,32 @@ def test_02_admin_checkout(self):
...
+
+@odoo.tests.tagged('post_install', '-at_install')
+class TestWebsiteSaleCheckoutAddress(odoo.tests.TransactionCase):
+    def test_01_edit_billing_address(self):
+        website = self.env['website'].browse(1)
+        p = self.env.user.partner_id
+        so = self.env['sale.order'].create({
+            'partner_id': p.id,
+            'website_id': website.id,
+        })
```
Comment (JKE-be) [inline addons/website_sale/tests/test_sale_process.py]: order_line: [(0, 0, { vals })]

Comment (rdeodoo) [reply addons/website_sale/tests/test_sale_process.py]: Done

File: addons/website_sale/tests/test_sale_process.py (hunk 1 above)
Comment (rdeodoo) [reply addons/website_sale/tests/test_sale_process.py]: Done

**Comment by JKE-be** [comment]
@robodoo r+

**Comment by JKE-be** [comment]
@robodoo rebase-merge

**Pull request closed without merging**